app/controllers/samples_controller.rb

```diff
@@ -174,18 +174,21 @@ def show
     background_id = check_background_id(@sample)
     @report_page_params = { pipeline_version: @pipeline_version, background_id: background_id } if background_id
     @report_page_params[:scoring_model] = params[:scoring_model] if params[:scoring_model]
-    if @pipeline_run && (((@pipeline_run.adjusted_remaining_reads.to_i > 0 || @pipeline_run.results_finalized?) && !@pipeline_run.failed?) || @pipeline_run.report_ready?)
-      if background_id
-        @report_present = 1
-        @report_ts = @pipeline_run.updated_at.to_i
-        @all_categories = all_categories
-        @report_details = report_details(@pipeline_run, current_user.id)
-        @ercc_comparison = @pipeline_run.compare_ercc_counts
-      end
 
-      if @pipeline_run.failed?
-        @pipeline_run_retriable = true
-      end
+    # Report is ready if pipeline run has results finalized and did not fail
+    # (success), or if all its output states are loaded successfully.
+    if @pipeline_run &&
```

Comment:
So, report can be ready event if the `report_ready` flag is not set?

Reply:
Actually it doesn't look like it anymore

Reply:
Oh wait actually this is so that some "COMPLETE*" reports that failed post processing can still be loaded.. report_ready? is only true if all output states succeed so it's hacky but the COMPLETE* ones will still load with results_finalized and !failed..